notifications_utils/logging.py

```diff
@@ -101,57 +95,44 @@ def ensure_log_path_exists(path):
     This function assumes you're passing a path to a file and attempts to create
     the path leading to that file.
     """
+
     try:
         Path(path).parent.mkdir(mode=755, parents=True)
     except FileExistsError:
+        # The path and file already exist, which is acceptable.
         pass
 
 
-def get_handlers(app):
-    handlers = []
-    standard_formatter = CustomLogFormatter(LOG_FORMAT, TIME_FORMAT)
-    json_formatter = JSONFormatter(LOG_FORMAT, TIME_FORMAT)
+def is_200_static_log(log) -> bool:
+    """ Turn off 200 OK static logs in development. """
 
-    stream_handler = logging.StreamHandler(sys.stdout)
-    if not app.debug:
-        # machine readable json to both file and stdout
-        # file_handler = logging.handlers.WatchedFileHandler(
-        #     filename='{}.json'.format(app.config['NOTIFY_LOG_PATH'])
-        # )
-
-        handlers.append(configure_handler(stream_handler, app, json_formatter))
-        # Do not write to files, stdout logging is only needed
-        # handlers.append(configure_handler(file_handler, app, json_formatter))
-    else:
-        # turn off 200 OK static logs in development
-        def is_200_static_log(log):
-            msg = log.getMessage()
-            return not ('GET /static/' in msg and ' 200 ' in msg)
+    msg = log.getMessage()
+    return not ('GET /static/' in msg and ' 200 ' in msg)
 
-        logging.getLogger('werkzeug').addFilter(is_200_static_log)
-
-        # human readable stdout logs
-        handlers.append(configure_handler(stream_handler, app, standard_formatter))
-
-    return handlers
 
+def get_handler(app):
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setLevel(logging.getLevelName(app.config['NOTIFY_LOG_LEVEL']))
+    stream_handler.addFilter(AppNameFilter(app.config['NOTIFY_APP_NAME']))
+    stream_handler.addFilter(RequestIdFilter())
 
-def configure_handler(handler, app, formatter):
-    handler.setLevel(logging.getLevelName(app.config['NOTIFY_LOG_LEVEL']))
-    handler.setFormatter(formatter)
-    handler.addFilter(AppNameFilter(app.config['NOTIFY_APP_NAME']))
-    handler.addFilter(RequestIdFilter())
+    if app.debug:
+        # Human readable stdout logs that omit static route 200 responses
+        logging.getLogger('werkzeug').addFilter(is_200_static_log)
+        stream_handler.setFormatter(logging.Formatter(LOG_FORMAT, TIME_FORMAT))
+    else:
+        # Machine readable json to stdout
+        stream_handler.setFormatter(JsonFormatter(LOG_FORMAT, TIME_FORMAT))
```

Comment:
Why logging formatter if in debug but json formatter if not in debug?

Reply:
I preserved the existing behavior.  I think the intent is not to clutter your shell when running locally and to use a machine-parsable format with CloudWatch.